### PR TITLE
RATEST-200: E2E Location management workflow test draft

### PR DESCRIPTION
**Pull request by Ndacyayisenga-droid**
RATEST-200: E2E Location management workflow test draft

cc @kdaud  @sherrif10 
I need some advice here

### CODE REVIEW

File: src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    public void visitHomePage() {
+        initiateWithLogin();
+    }
+
+    @After(RunTest.HOOK.SELENIUM_LOCATION_MANAGEMENT)
+    public void destroy() {
+        quit();
+    }
+
+
```
Comment (mherman22) [inline src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java:63]: great work,please correct the indention here,

**Review by mherman22 — CHANGES_REQUESTED** [review]

File: src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+import static org.junit.Assert.assertTrue;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.openmrs.contrib.qaframework.RunTest;
+
+
```
Comment (kdaud) [inline src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java:21]: any reason for having two empty lines here?

File: src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+import org.openmrs.contrib.qaframework.page.ManageLocationAttributeTypesPage;
+import org.openmrs.contrib.qaframework.page.LocationPage;
+import org.openmrs.contrib.qaframework.page.ManageLocationsPage;
+import org.openmrs.contrib.qaframework.page.AddEditLocationPage;
+import org.openmrs.contrib.qaframework.page.ManageLocationTagsPage;
+import org.openmrs.contrib.qaframework.page.ConfigureMetadataPage;
+
+
```
Comment (kdaud) [inline src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java:29]: even here!

File: src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    private static final String STATE = "Central";
+    private static final String COUNTRY = "Uganda";
+    private ConfigureMetadataPage configureMetadataPage;
+    private LocationPage locationPage;
+    private ManageLocationsPage manageLocationsPage;
+    private AddEditLocationPage addEditLocationPage;
+    private ManageLocationTagsPage manageLocationTagsPage;
+    private ManageLocationAttributeTypesPage manageLocationAttributeTypesPage;
+
+
```
Comment (kdaud) [inline src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java:52]: also here!

File: src/test/resources/features/refapp-2.x/05-location/locations.feature [hunk 5]
```diff
@@ -1 +1,29 @@
...
+    When a user clicks on manage location tag
+    Then the user clicks on add new location
+    And a user fills add new location tag form
+    And a user saves add New location tag form
+    Then the New location tag form is saved
+
+    # User story: Manager locations
+    When a user clicks on manager locations
+    Then the user click on add new location
+    And a user fills add new location form
```
Comment (kdaud) [inline src/test/resources/features/refapp-2.x/05-location/locations.feature:29]: did you forget to add an empty line to cater for git collaborators?

**Comment by mherman22** [comment]
you forgot to add a script for the location management workflow test in the[package.json](https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/package.json)

File: src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+        initiateWithLogin();
+    }
+
+    @After(RunTest.HOOK.SELENIUM_LOCATION_MANAGEMENT)
+    public void destroy() {
+        quit();
+    }
+
+
+    @When("a user clicks on the configure metadata link from the home page")
```
Comment (mherman22) [inline src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java:64]: your [feature file step here](https://github.com/openmrs/openmrs-contrib-qaframework/blob/6d3e1f1e1b6ed7c369ae818561cea3d69dd5eba7/src/test/resources/features/refapp-2.x/05-location/locations.feature#L4) doesnot correspond with this step....
you have `When a user clicks on configure metadata link from the home page` in the feature unlike in your steps where you have `a user clicks on the configure metadata link from the home page`..it must be identical

**Review by mherman22 — CHANGES_REQUESTED** [review]

File: src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java (hunk 6 above)
Comment (mherman22) [reply src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java:64]: walk through your feature file and ensure the steps there are identical to the ones you have in you step definitions

File: src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+        manageLocationAttributeTypesPage.goToAddNewLocationAttributeTypeForm();
+    }
+
+    @And("a user fills the form")
+    public void fillsTheForm(){
+        manageLocationAttributeTypesPage.enterLocationAttributeTypeName(NAME);
+        manageLocationAttributeTypesPage.enterLocationAttributeTypeDescription(DESCRIPTION);
+        manageLocationAttributeTypesPage.enterMinimumOccurs(MIN_VALUE);
+        manageLocationAttributeTypesPage.enterMaximumOccurs(MAX_VALUE);
+        manageLocationAttributeTypesPage.selectDatatype("Location Datatype");
```
Comment (mherman22) [inline src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java:88]: the element here (line 88) is causing this build failure as shown in this [cucumber report](https://reports.cucumber.io/reports/98c224ae-483b-4c9e-a35f-9c6abc85fc57). take a look and ensure the cssSelector is up-to-date with the newly released refapp version

Comment (Ndacyayisenga-droid) [reply src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java:88]: > 

@mherman22 I tried to change it to an id to see what will though the cssSelector is upto date

Comment (Ndacyayisenga-droid) [inline src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java:88]: Hi @kdaud, I have a problem with the method that implements this functionality

File: src/test/resources/features/refapp-2.x/05-location/locations.feature [hunk 8]
```diff
@@ -1 +1,28 @@
-Feature: Location Management
\ No newline at end of file
+Feature: Location Management
+
+  Background:
+
+
```
Comment (kdaud) [inline src/test/resources/features/refapp-2.x/05-location/locations.feature:5]: what is the background of the user stories?

File: src/test/resources/features/refapp-2.x/05-location/locations.feature (hunk 5 above)
Comment (kdaud) [reply src/test/resources/features/refapp-2.x/05-location/locations.feature:28]: @Ndacyayisenga-droid did you address the comment [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/242/files#r756006352)?

File: src/test/java/org/openmrs/contrib/qaframework/page/ManageLocationAttributeTypesPage.java [hunk 9]
```diff
@@ -20,7 +20,8 @@
 	private static final By ATTRIBUTE_TYPE_DESCRIPTION_FIELD = By.cssSelector("#description-field");
 	private static final By MIN_OCCURS_FIELD = By.cssSelector("#minOccurs-field");
 	private static final By MAX_OCCURS_FIELD = By.cssSelector("#maxOccurs-field");
-	private static final By DATA_TYPE_FIELD = By.cssSelector("#datatypeClassname-field");
+	//private static final By DATA_TYPE_FIELD = By.cssSelector("#datatypeClassname-field");
```
Comment (sherrif10) [inline src/test/java/org/openmrs/contrib/qaframework/page/ManageLocationAttributeTypesPage.java:23]: Any reason why this is commented.

**Comment by kdaud** [comment]
@Ndacyayisenga-droid any progress with this workflow test?

**Comment by Ndacyayisenga-droid** [comment]
cc @kdaud

File: src/test/resources/org/openmrs/uitestframework/test-local.properties [hunk 10]
```diff
@@ -1,9 +1,8 @@
 webapp.url=http://localhost:8080/openmrs
-login.username=admin
```
Comment (sherrif10) [inline src/test/resources/org/openmrs/uitestframework/test-local.properties]: Any reason for removing this property

File: src/test/resources/features/refapp-2.x/05-location/locations.feature (hunk 8 above)
Comment (sherrif10) [reply src/test/resources/features/refapp-2.x/05-location/locations.feature:6]: Did you see this comment

File: src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java [hunk 11]
```diff
@@ -292,3 +292,4 @@ public void systemEndsPatientVisit() {
 		dashboardPage = visitsDashboardPage.goToPatientDashboard();
 	}
 }
+
```
Comment (sherrif10) [inline src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java:295]: Why adding a new line at the end of this class

File: src/test/resources/features/refapp-2.x/05-location/locations.feature (hunk 5 above)
Comment (sherrif10) [reply src/test/resources/features/refapp-2.x/05-location/locations.feature:28]: Feel free to address all the comments being reviewed. Thanks

**Comment by kdaud** [comment]
@Ndacyayisenga-droid some merge conflicts.

**Pull request closed without merging**